crates/bevy_ecs/src/schedule/executor.rs

```diff
@@ -31,6 +31,10 @@ impl ParallelSystemExecutor for SingleThreadedExecutor {
 
         for system in systems {
             if system.should_run() {
+                #[cfg(feature = "trace")]
+                let system_span = bevy_utils::tracing::info_span!("system", name = &*system.name());
+                #[cfg(feature = "trace")]
+                let _system_guard = system_span.enter();
```

Comment:
You can collapse these into a single block statement so you only need one use of `#[cfg(...)]`
```rust
#[cfg(feature = "trace")] {
    let system_span = bevy_utils::tracing::info_span!("system", name = &*system.name());
    let _system_guard = system_span.enter();
}
```

Reply:
this way `_system_guard` is dropped with the block which ends the span

Reply:
Ahh oh yea, I forgot spans have drop impls!